samplesheets/views_api.py

```diff
@@ -310,6 +323,240 @@ def post(self, request, *args, **kwargs):
+
+
+class IrodsAccessTicketCreateAPIView(
+    IrodsAccessTicketModifyMixin, SODARAPIBaseProjectMixin, CreateAPIView
+):
+    """
+    Create an iRODS access ticket for a project.
+
+    **URL:** ``/samplesheets/api/irods/ticket/create/{Project.sodar_uuid}``
+
+    **Methods:** ``POST``
+
+    **Parameters:**
+
+    - ``assay``: Assay primary key
+    - ``path``: iRODS path
+    - ``label``: Label (optional)
+    - ``date_expires``: Expiration date (YYYY-MM-DDThh:mm:ssZ, optional)
+
+    **Returns:**
+
+    - ``ticket``: iRODS access ticket
+    """
+
+    permission_required = 'samplesheets.edit_sheet'
+
+    def post(self, request, *args, **kwargs):
+        """POST request for creating an iRODS access ticket"""
+        irods_backend = get_backend_api('omics_irods')
+        self.project = self.get_project()
+        serializer = IrodsAccessTicketSerializer(
+            data=request.data, partial=True
+        )
+        serializer.context['project'] = self.project
+        serializer.context['user'] = self.request.user
+
+        if serializer.is_valid():
+            try:
+                with irods_backend.get_session() as irods:
+                    ticket = self.create_irods_ticket(
+                        irods_backend,
+                        irods,
+                        serializer.validated_data.get('path'),
+                        build_secret(16),
+                        serializer.validated_data.get('date_expires'),
+                    )
+            except Exception as ex:
+                raise ValidationError(
+                    '{} {}'.format('Creating ' + IRODS_TICKET_EX_MSG + ':', ex)
+                )
+
+            # Create database object
+            serializer.validated_data['ticket'] = ticket.ticket
+            irods_access_ticket = serializer.save()
+
+            # Create timeline event and app alerts
+            self.create_timeline_event(irods_access_ticket, 'create')
+            self.create_app_alerts(
+                irods_access_ticket, 'create', self.request.user
+            )
+        else:
+            raise ValidationError('Invalid data: {}'.format(serializer.errors))
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+class IrodsAccessTicketUpdateAPIView(
+    IrodsAccessTicketModifyMixin, SODARAPIBaseProjectMixin, UpdateAPIView
+):
+    """
+    Update an iRODS access ticket for a project.
+
+    **URL:** ``/samplesheets/api/irods/ticket/update/{IrodsAccessTicket.sodar_uuid}``
+
+    **Methods:** ``PUT``, ``PATCH``
+
+    **Parameters:**
+
+    - ``label``: Label
+    - ``date_expires``: Expiration date (YYYY-MM-DD)
+
+    **Returns:**
+
+    - ``ticket``: iRODS access ticket
+    """
+
+    permission_required = 'samplesheets.edit_sheet'
+
+    def get_object(self):
+        return IrodsAccessTicket.objects.get(
+            sodar_uuid=self.kwargs.get('irodsaccessticket')
+        )
+
+    def put(self, request, *args, **kwargs):
```

Comment:
Instead of `put()`, we should be overriding `perform_update()`. See `IrodsDataRequestUpdateAPIView`. Also, if you only override `put()`, `PATCH` requests will not be correctly handled.